Frontend/src/ui/App.jsx

```diff
@@ -1,23 +1,44 @@
-import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import {
+  QueryClient,
+  QueryClientProvider,
+  useQuery,
+} from '@tanstack/react-query'
 import { ReactQueryDevtools } from '@tanstack/react-query-devtools'
-import React, { useCallback, useState } from 'react'
-import { AuthContext } from '../api/helper/context/auth_context'
+import React from 'react'
+import { getPermissions } from '../api/auth/get_permissions'
+import { PermissionsContext } from '../api/helper/context/permission_context'
+import { UserContext } from '../api/helper/context/user_context'
+import getMe from '../api/user/get/get_me'
+import LoadingMessage from './messages/loadingMessage'
 
 const queryClient = new QueryClient()
 export const USER_KEY = 'user'
 export const JWT_KEY = 'jwt'
 export default function App({ children }) {
-  const [user, setUserState] = useState(localStorage.getItem(USER_KEY))
-  const setUser = useCallback((user) => {
-    localStorage.setItem(USER_KEY, user)
-    localStorage.removeItem(JWT_KEY)
-    setUserState(user)
-  }, [])
+  const { isLoading, data: user } = useQuery({
+    queryKey: ['user-me'],
+    queryFn: () => getMe(),
+  })
+  const { isLoading: permissionsLoading, data: permissions } = useQuery({
```

Comment:
There should be the "Provider" wrapper component that does this fetch.

You can reference my AuthProvider here: https://github.com/coder13/delegateDashboard/blob/main/src/providers/AuthProvider.js